packages/x-charts/src/Legend/legendClasses.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  unstable_generateUtilityClass as generateUtilityClass,
+  unstable_generateUtilityClasses as generateUtilityClasses,
+} from '@mui/utils';
+
+export interface LegendClasses {
+  /** Styles applied to the root element. */
+  root: string;
+  /** Styles applied to series element. */
+  series: string;
+  /** Styles ap plied to series mark element. */
+  mark: string;
+  /** Styles applied to series label. */
+  label: string;
+  /** Styles applied to the legend with column layout. */
+  column: string;
+  /** Styles applied to the legend with row layout. */
+  row: string;
+}
+
+export function getLegendUtilityClass(slot: string) {
+  return generateUtilityClass('MuiLegend', slot);
+}
+
+export const legendClasses: LegendClasses = generateUtilityClasses('MuiLegend', [
```

Comment:
I was doing a mix with `Charts` added to the `styled()` name but not to the `useThemeProps()` name.

I went the other way: remove all the `Charts` from `styled()` name

Reply:
I'm not sure it is for the best... 🤔 
On the pickers, we do opt for exclusivity.
Now you'd have `MuiTooltip` on the charts tooltip, but material ui also has a tooltip component and so, styles between those classes could clash given a certain way of providing overridden styles. 🙈 🤷

Reply:
> On the pickers, we do opt for exclusivity.

We add a picker but only for sub components. For example, you have `MuiPickersCalendarHeader` but not `MuiPickersDigitalClock`. It's `MuiDigitalClock`.

The lint rules enforce that `useThemeProps({name: "MuiChartsLegend" })` implies the component name is `ChartsLegend`.
Going in that direction, all the components will be `<ChartsLegend />`, `<ChartsXAxis />`, `<ChartsTooltip />`, ...

For the Tooltip, it's not a problem. It already exists core, and we need to differentiate them. But for others, it seems boring to always have those `Charts` added.

By the way, I just noticed that `@mui/joy` is prefixing all its components with `Joy`. Maybe it would just make sens to prefic names with `Charts` without the `Mui`

Reply:
For example in Sheet component: https://github.com/mui/material-ui/blob/4107bdbc8676153ee71aedc7f6388070cb14ac53/packages/mui-joy/src/Sheet/Sheet.tsx

Reply:
That's interesting... Because in the docs the rendered class is still `MuiButton` and not `JoyButton` as you'd expect by that name. 🤔

Reply:
@siriwatknp Could you share some details about this topic?

For example, the `<Sheet />` components get classes prefixed with `Mui`

https://github.com/mui/material-ui/blob/4107bdbc8676153ee71aedc7f6388070cb14ac53/packages/mui-joy/src/Sheet/sheetClasses.ts#L32-L37
```jsx
export function getSheetUtilityClass(slot: string): string {
  return generateUtilityClass('MuiSheet', slot);
}

const sheetClasses: SheetClasses = generateUtilityClasses('MuiSheet', [
  'root',
  ....
]);
```

But the component style override is using `Joy` prefix
https://github.com/mui/material-ui/blob/4107bdbc8676153ee71aedc7f6388070cb14ac53/packages/mui-joy/src/Sheet/Sheet.tsx
```jsx
export const SheetRoot = styled('div', {
  name: 'JoySheet',
  slot: 'Root',
  overridesResolver: (props, styles) => styles.root,
})<{ ownerState: SheetOwnerState }>(({ theme, ownerState }) => {
 // ...
 
 
const Sheet = React.forwardRef(function Sheet(inProps, ref) {
  const props = useThemeProps<typeof inProps & SheetProps>({
    props: inProps,
    name: 'JoySheet',
  });
  // ...
```

Is it to overcome some specific issue, or a standard you built?

Reply:
That is totally not a problem and probably even better/cleaner if we feel that we want to go prefixing all the components. 👌 👍

Reply:
@alexfauquette the decision about the classname is in https://github.com/mui/material-ui/issues/33260#issuecomment-1184426367.

However, for `styleOverrides`, we don't have any conclusion yet because developers are using both Joy UI and Material UI together. If we make the component identifier to be the same, e.g. `MuiButton`, it will be really hard to customize.